doc/api/worker_threads.md

```diff
@@ -274,6 +274,109 @@ if (isMainThread) {
 }
 ```
 
+## `worker.locks`
+<!-- YAML
+added: REPLACEME
+-->
+
+> Stability: 1 - Experimental
+
+* {LockManager}
+
+An instance of a [`LockManager`][].
+
+### Class: `Lock`
+<!-- YAML
+added: REPLACEME
+-->
+
+The Lock interface provides the name and mode of a previously requested lock,
+which is received in the callback to [`locks.request()`][].
+
+#### `lock.name`
+<!-- YAML
+added: REPLACEME
+-->
+
+* {string}
+
+The name of this lock.
+
+#### `lock.mode`
+<!-- YAML
+added: REPLACEME
+-->
+
+* {string}
+
+The mode of this lock. Either `shared` or `exclusive`.
+
+### Class: `LockManager`
+<!-- YAML
+added: REPLACEME
+-->
```

Comment:
While I know that the Web Locks API itself uses `LockManager` as a singleton, I've been thinking that it may be interesting to allow users to create multiple `LockManager` instances in case they want to avoid relying on a global lock namespace.

```js
const { createLockManager, Worker } = require('workers');

const lockManager = createLockManager('namespace-root');
const { port1, port2 } = new MessageChannel();
new Worker('worker.js', { workerData: lockManager });
```

Reply:
I think that shouldn’t be too hard theoretically, but since it would be non-standard, I’d also not implement this at this point. I’d expect that namespacing takes place anyway in practice, even with the singleton variant.